src/vs/workbench/contrib/chat/browser/chatContentParts/chatMarkdownContentPart.ts

```diff
@@ -355,58 +355,60 @@ class CollapsedCodeBlock extends Disposable {
 
 		this._uri = uri;
 
+		const session = this.chatService.getSession(this.sessionId);
 		const iconText = this.labelService.getUriBasenameLabel(uri);
-		const editSession = this.chatEditingService.getEditingSession(this.sessionId);
-		const modifiedEntry = editSession?.getEntry(uri);
-		const isComplete = !modifiedEntry?.isCurrentlyBeingModifiedBy.get();
-
-		let iconClasses: string[] = [];
-		if (isStreaming || !isComplete) {
-			const codicon = ThemeIcon.modify(Codicon.loading, 'spin');
-			iconClasses = ThemeIcon.asClassNameArray(codicon);
-		} else {
-			const fileKind = uri.path.endsWith('/') ? FileKind.FOLDER : FileKind.FILE;
-			iconClasses = getIconClasses(this.modelService, this.languageService, uri, fileKind);
-		}
-
-		const iconEl = dom.$('span.icon');
-		iconEl.classList.add(...iconClasses);
+		this._progressStore.add(autorun(r => {
+			const editSession = session?.editingSessionObs?.promiseResult.read(r)?.data;
```

Comment:
Not an observables expert, does this look ok @connor4312?

Reply:
Nvm, this caused some rendering issues and I changed it again in https://github.com/microsoft/vscode/pull/244398. Now it's not so different from the original state.